### PR TITLE
[Validator] Edit and add missing translations for Persian (fa)

### LINKED ISSUE #43490: Missing translations for Persian (fa)
Hello,

There are some missing translation for Persian (fa). These should be added in branch 4.4. 
See https://github.com/symfony/symfony/issues/38710 for details and [this page](https://symfony-translations.nyholm.tech/#pr) for an example.

These are the files that should be updated: 
- [src/Symfony/Component/Validator/Resources/translations/validators.fa.xlf](https://github.com/symfony/symfony/blob/4.4/src/Symfony/Component/Validator/Resources/translations/validators.fa.xlf)

NOTE: If you want to work on this issue, add a comment to assign it to yourself and let others know that this is already taken.

### PR DESCRIPTION
**Pull request by BahmanMehrdad**
| Q             | A
| ------------- | ---
| Branch?       | 4.4
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Tickets       | Fix #43490
| License       | MIT

This PR is for edit and add missing translations for Persian (fa)

### CODE REVIEW

File: src/Symfony/Component/Validator/Resources/translations/validators.fa.xlf
Comment (ErfanBahramali) [inline src/Symfony/Component/Validator/Resources/translations/validators.fa.xlf:55]: This is not wrong, but there is no reason for this change and I do not see the need to do so

File: src/Symfony/Component/Validator/Resources/translations/validators.fa.xlf
Comment (ErfanBahramali) [inline src/Symfony/Component/Validator/Resources/translations/validators.fa.xlf]: This is charset, not character

This is wrong

File: src/Symfony/Component/Validator/Resources/translations/validators.fa.xlf
Comment (ErfanBahramali) [inline src/Symfony/Component/Validator/Resources/translations/validators.fa.xlf:339]: This is not wrong, but there is no reason for this change and I do not see the need to do so

File: src/Symfony/Component/Validator/Resources/translations/validators.fa.xlf
Comment (ErfanBahramali) [inline src/Symfony/Component/Validator/Resources/translations/validators.fa.xlf]: This is wrong

File: src/Symfony/Component/Validator/Resources/translations/validators.fa.xlf
Comment (ErfanBahramali) [inline src/Symfony/Component/Validator/Resources/translations/validators.fa.xlf:371]: I don't think this translation is correct

**Comment by ErfanBahramali** [comment]
Converting half-space (semi-space) to space is wrong

**Comment by BahmanMehrdad** [comment]
No half-space (semi-space) converted to space !!!!
Did you check the file changed?

File: src/Symfony/Component/Validator/Resources/translations/validators.fa.xlf
Comment (ErfanBahramali) [inline src/Symfony/Component/Validator/Resources/translations/validators.fa.xlf:387]: I don't agree with this change

**Review by ErfanBahramali — CHANGES_REQUESTED** [review]

File: src/Symfony/Component/Validator/Resources/translations/validators.fa.xlf
Comment (BahmanMehrdad) [reply src/Symfony/Component/Validator/Resources/translations/validators.fa.xlf]: Charset means character set. I add set to this translation

Comment (ErfanBahramali) [reply src/Symfony/Component/Validator/Resources/translations/validators.fa.xlf]: it's wrong
You just changed it to finglish and did not translate

The previous translation is correct
No need to change

File: src/Symfony/Component/Validator/Resources/translations/validators.fa.xlf
Comment (ErfanBahramali) [reply src/Symfony/Component/Validator/Resources/translations/validators.fa.xlf:55]: The previous translation is correct
No need to change

File: src/Symfony/Component/Validator/Resources/translations/validators.fa.xlf
Comment (ErfanBahramali) [reply src/Symfony/Component/Validator/Resources/translations/validators.fa.xlf:339]: The previous translation is correct
No need to change

File: src/Symfony/Component/Validator/Resources/translations/validators.fa.xlf
Comment (ErfanBahramali) [reply src/Symfony/Component/Validator/Resources/translations/validators.fa.xlf]: The previous translation is correct
No need to change

File: src/Symfony/Component/Validator/Resources/translations/validators.fa.xlf
Comment (ErfanBahramali) [reply src/Symfony/Component/Validator/Resources/translations/validators.fa.xlf:387]: The previous translation is correct
No need to change

File: src/Symfony/Component/Validator/Resources/translations/validators.fa.xlf
Comment (ErfanBahramali) [reply src/Symfony/Component/Validator/Resources/translations/validators.fa.xlf:371]: The previous translation is correct
No need to change

File: src/Symfony/Component/Validator/Resources/translations/validators.fa.xlf
Comment (BahmanMehrdad) [reply src/Symfony/Component/Validator/Resources/translations/validators.fa.xlf:55]: "رایانامه" is not common in Persian

File: src/Symfony/Component/Validator/Resources/translations/validators.fa.xlf
Comment (BahmanMehrdad) [reply src/Symfony/Component/Validator/Resources/translations/validators.fa.xlf]: In Persian we say "کاراکترست". This translation is for Persian languages and Finglish does not make sense

File: src/Symfony/Component/Validator/Resources/translations/validators.fa.xlf
Comment (BahmanMehrdad) [reply src/Symfony/Component/Validator/Resources/translations/validators.fa.xlf:339]: "حاوی" is wrong and "شامل" is correct. And "منحصر به فرد" is better than "یکتا"

File: src/Symfony/Component/Validator/Resources/translations/validators.fa.xlf
Comment (BahmanMehrdad) [reply src/Symfony/Component/Validator/Resources/translations/validators.fa.xlf]: The previous translation is incorrect and does not match the English text

File: src/Symfony/Component/Validator/Resources/translations/validators.fa.xlf
Comment (BahmanMehrdad) [reply src/Symfony/Component/Validator/Resources/translations/validators.fa.xlf:371]: "hostname" in Persian means "نام میزبان" and must be translated, but it was not translated in the previous translation and its English word is inserted.

File: src/Symfony/Component/Validator/Resources/translations/validators.fa.xlf
Comment (BahmanMehrdad) [reply src/Symfony/Component/Validator/Resources/translations/validators.fa.xlf:387]: The previous translation was incomplete and did not convey the exact meaning

**Review by BahmanMehrdad — COMMENTED** [review]
Review

File: src/Symfony/Component/Validator/Resources/translations/validators.fa.xlf
Comment (ErfanBahramali) [reply src/Symfony/Component/Validator/Resources/translations/validators.fa.xlf]: This is wrong

File: src/Symfony/Component/Validator/Resources/translations/validators.fa.xlf
Comment (ErfanBahramali) [reply src/Symfony/Component/Validator/Resources/translations/validators.fa.xlf:339]: This is wrong

File: src/Symfony/Component/Validator/Resources/translations/validators.fa.xlf
Comment (BahmanMehrdad) [reply src/Symfony/Component/Validator/Resources/translations/validators.fa.xlf]: See this:
https://www.w3schools.com/html/html_charset.asp

File: src/Symfony/Component/Validator/Resources/translations/validators.fa.xlf
Comment (BahmanMehrdad) [reply src/Symfony/Component/Validator/Resources/translations/validators.fa.xlf:339]: Are you familiar with the principles and rules of Persian language?

File: src/Symfony/Component/Validator/Resources/translations/validators.fa.xlf
Comment (ErfanBahramali) [reply src/Symfony/Component/Validator/Resources/translations/validators.fa.xlf]: Your translation is not accurate
correct:

این رمزعبور در یک رخنه‌ی اطلاعاتی نشت کرده است، نباید از آن استفاده کرد. لطفاً از رمزعبور دیگری استفاده کنید.

Comment (BahmanMehrdad) [reply src/Symfony/Component/Validator/Resources/translations/validators.fa.xlf]: "it must not be used" means "نباید از آن استفاده کرد", and this sentence did not exist in the previous translation

**Comment by ErfanBahramali** [comment]
It is as if there is disagreement and there is a need to discuss these issues

@BahmanMD 
Contact me on Telegram or Discord
Or you can raise issues in the topic so that other friends can participate

**Comment by BahmanMehrdad** [comment]
> It is as if there is disagreement and there is a need to discuss these issues
> 
> @BahmanMD Contact me on Telegram or Discord Or you can raise issues in the topic so that other friends can participate

Unfortunately, most of the objections to the translation indicate that you are not fluent in Persian, because the meanings of the words in Persian may be the same, but the time and place of their use are very important.

File: src/Symfony/Component/Validator/Resources/translations/validators.fa.xlf
Comment (ErfanBahramali) [reply src/Symfony/Component/Validator/Resources/translations/validators.fa.xlf]: Adding that section is correct, but the first and last item is incorrect
Means 'است' and 'دیگر'
The first case was corrected ( اس ) => ( است )
But the second 
' لطفاً از رمزعبور دیگری استفاده کنید' is correct

Comment (BahmanMehrdad) [reply src/Symfony/Component/Validator/Resources/translations/validators.fa.xlf]: "This password has been leaked in a data breach" means "این  رمزعبور در یک رخنه‌ی اطلاعاتی نشت کرده است"
"it must not be used" means "نباید از آن استفاده کرد"
"Please use another password" means "لطفا از یک رمزعبور دیگری استفاده کنید"
Why is this translation wrong !!!!!?

**Comment by ErfanBahramali** [comment]
> > It is as if there is disagreement and there is a need to discuss these issues
> > @BahmanMD Contact me on Telegram or Discord Or you can raise issues in the topic so that other friends can participate
> 
> Unfortunately, most of the objections to the translation indicate that you are not fluent in Persian, because the meanings of the words in Persian may be the same, but the time and place of their use are very important.

I just say my opinion
In this case, you can also get help from the previous translator ( Before me )

File: src/Symfony/Component/Validator/Resources/translations/validators.fa.xlf
Comment (ErfanBahramali) [reply src/Symfony/Component/Validator/Resources/translations/validators.fa.xlf]: 'یک' is extra

"Please use another password" means "لطفا از رمزعبور دیگری استفاده کنید"

File: src/Symfony/Component/Validator/Resources/translations/validators.fa.xlf
Comment (ErfanBahramali) [reply src/Symfony/Component/Validator/Resources/translations/validators.fa.xlf]: 'کاراکترست' is not a Persian word
You just converted it to Finglish

File: src/Symfony/Component/Validator/Resources/translations/validators.fa.xlf
Comment (ErfanBahramali) [reply src/Symfony/Component/Validator/Resources/translations/validators.fa.xlf:55]: Because 'پست الکترونیک' is English and 'رایانامه' is the official translation of 'email'

**Comment by BahmanMehrdad** [comment]
> > > It is as if there is disagreement and there is a need to discuss these issues
> > > @BahmanMD Contact me on Telegram or Discord Or you can raise issues in the topic so that other friends can participate
> > 
> > 
> > Unfortunately, most of the objections to the translation indicate that you are not fluent in Persian, because the meanings of the words in Persian may be the same, but the time and place of their use are very important.
> 
> I just say my opinion In this case, you can also get help from the previous translator ( Before me )

Thank you for your opinion. Considering that my main language is Persian and also the previous translations seem to have been translated by Google Translate, it is better to translate them fluently and slang.

File: src/Symfony/Component/Validator/Resources/translations/validators.fa.xlf
Comment (ErfanBahramali) [reply src/Symfony/Component/Validator/Resources/translations/validators.fa.xlf:339]: 'حاوی' is not wrong

'یکتا' is a simpler and shorter word

No need to change

File: src/Symfony/Component/Validator/Resources/translations/validators.fa.xlf
Comment (BahmanMehrdad) [reply src/Symfony/Component/Validator/Resources/translations/validators.fa.xlf:55]: It is exactly like saying "بالگرد" instead of "هلیکوپتر", which is never expressed in this way in fluent Persian. You know that "رایانامه" is only used for some office correspondence and is by no means popular

**Comment by ErfanBahramali** [comment]
> > > > It is as if there is disagreement and there is a need to discuss these issues
> > > > @BahmanMD Contact me on Telegram or Discord Or you can raise issues in the topic so that other friends can participate
> > > 
> > > 
> > > Unfortunately, most of the objections to the translation indicate that you are not fluent in Persian, because the meanings of the words in Persian may be the same, but the time and place of their use are very important.
> > 
> > 
> > I just say my opinion In this case, you can also get help from the previous translator ( Before me )
> 
> Thank you for your opinion. Considering that my main language is Persian and also the previous translations seem to have been translated by Google Translate, it is better to translate them fluently and slang.

I declare what I think is right according to the documents

File: src/Symfony/Component/Validator/Resources/translations/validators.fa.xlf
Comment (BahmanMehrdad) [reply src/Symfony/Component/Validator/Resources/translations/validators.fa.xlf]: Finglish is not our official language and our official language is Persian, the meaning of this word is correct in the form of "کاراکترست" and the writing in Finglish has no meaning.

File: src/Symfony/Component/Validator/Resources/translations/validators.fa.xlf
Comment (BahmanMehrdad) [reply src/Symfony/Component/Validator/Resources/translations/validators.fa.xlf:339]: "حاوی" and "شامل" are semantically different.
"منحصر به فرد" means smoother than "یکتا".
It is better to change them for a proper translation.

**Comment by BahmanMehrdad** [comment]
> > > > > It is as if there is disagreement and there is a need to discuss these issues
> > > > > @BahmanMD Contact me on Telegram or Discord Or you can raise issues in the topic so that other friends can participate
> > > > 
> > > > 
> > > > Unfortunately, most of the objections to the translation indicate that you are not fluent in Persian, because the meanings of the words in Persian may be the same, but the time and place of their use are very important.
> > > 
> > > 
> > > I just say my opinion In this case, you can also get help from the previous translator ( Before me )
> > 
> > 
> > Thank you for your opinion. Considering that my main language is Persian and also the previous translations seem to have been translated by Google Translate, it is better to translate them fluently and slang.
> 
> I declare what I think is right according to the documents

I guess you are not familiar with the rules and principles of the Persian language

**Comment by ErfanBahramali** [comment]
Your words are not convincing at all
I also mentioned the reason for every problem I reported

This discussion is useless

**Comment by ErfanBahramali** [comment]
It is better for more people to participate

Get help from friends who have worked on this translation before

**Comment by BahmanMehrdad** [comment]
Unfortunately, your unfamiliarity with the rules does not allow you to be convinced. I will also give you all the reasons I can give you to notice the flaw in the previous translation.
Exactly this discussion is useless.
Good luck

**Comment by BahmanMehrdad** [comment]
I would be happy for other friends to express their opinions.
Just remember that we want a fluent and meaningful translation and we do not want to use complicated words.

**Review by BahmanMehrdad — COMMENTED** [review]
Final review.

**Comment by BahmanMehrdad** [comment]
Hi @fabpot  and @derrabus 
I request to review this PR. Thank you

**Comment by fabpot** [comment]
I would love to get a review from another native speaker as none of the core team members is able to validate this PR.

File: src/Symfony/Component/Validator/Resources/translations/validators.fa.xlf
Comment (Arman-Hosseini) [reply src/Symfony/Component/Validator/Resources/translations/validators.fa.xlf:55]: I agree with @ErfanBahramali on this.

Comment (BahmanMehrdad) [reply src/Symfony/Component/Validator/Resources/translations/validators.fa.xlf:55]: @Arman-Hosseini 
آرمان جان ممنون از پیامت، کلمه رایانامه معمولا کاربردی نداره و کلمه سنگینی هست، بجز نامه های اداری من ندیدم جایی از این کلمه استفاده کنه که همون نامه های اداریهم از پست الکترونیکی استفاده میکنن. ما میخوایم ترجمه روان و راحتی داشته باشیم.
نظر شما و عرفان عزیز محترمه ولی بنظرم بهتره ترجمه عامیانه داشته باشیم تا کلماتی که در عموم تعریفی نداره. باز هم اگر شما دو بزرگوار فکر میکنین به رایانامه تغییر بدم بهتر هست اینکار رو انجام بدم.

Comment (BahmanMehrdad) [reply src/Symfony/Component/Validator/Resources/translations/validators.fa.xlf:55]: این کلمات بیشتر در نامه های آقای حداد عادل دیده میشه 😄 برای همین بنظرم استفاده از این سبک کلمات در فارسی روان اشتباه هست

**Comment by BahmanMehrdad** [comment]
Due to changes in the main file, this PR close and another PR open

**Pull request closed without merging**
